packages/generator/src/generator.ts

```diff
@@ -190,16 +190,19 @@ export async function generateProject(
   );
   await Promise.all(promises);
 
-  if (!options.serviceMapping) {
-    throw Error('The service mapping is undefined.');
-  }
+  if (options.optionsPerService) {
+    logger.verbose('Generating options per service ...');
 
-  project.createSourceFile(
-    resolve(options.serviceMapping.toString()),
-    serviceMappingFile(services),
-    { overwrite: true }
-  );
+    const dir = dirname(options.optionsPerService);
+    await mkdir(dir, { recursive: true });
```

Comment:
[q] What if the folder already exists? I guess covered by tests alrady.

Reply:
Made a quick test and got an error so please double check with a test 
![image](https://user-images.githubusercontent.com/56544999/214805410-501183b0-7adf-43e3-9055-77dbb45fae2e.png)

Reply:
I added few more tests, have a look

Reply:
the `recursive: true` makes the difference. No error.